code/game/machinery/constructable_frame.dm

```diff
@@ -445,6 +445,45 @@ to destroy them and players will be able to make replacements.
 	origin_tech = "programming=1"
 	req_components = list(
 							/obj/item/weapon/stock_parts/matter_bin = 1)
+	var/list/fridge_names_paths = list(
+							"\improper SmartFridge" = /obj/machinery/smartfridge,
+							"\improper MegaSeed Servitor" = /obj/machinery/smartfridge/seeds,
+							"\improper Refrigerated Medicine Storage" = /obj/machinery/smartfridge/medbay,
+							"\improper Slime Extract Storage" = /obj/machinery/smartfridge/secure/extract,
+							"\improper Secure Refrigerated Medicine Storage" = /obj/machinery/smartfridge/secure/medbay,
+							"\improper Smart Chemical Storage" = /obj/machinery/smartfridge/secure/chemistry,
+							"smart virus storage" = /obj/machinery/smartfridge/secure/chemistry/virology,
+							"\improper Drink Showcase" = /obj/machinery/smartfridge/drinks
+	)
+
+
+/obj/item/weapon/circuitboard/smartfridge/New()
+	..()
+	if(build_path == /obj/machinery/smartfridge)			//If we spawn the base type board (determined by the base type machine as the build path), become a random fridge board

```

Comment:
Remove this bit. Logic Gates have it because they have a non-constructable base type machine that the others inherit from. Smartfridges don't share that limitation, so you'd end up having the normal smartfridge in botany/kitchen containing a board for a random fridge type (sometimes the right one, sometimes the virus one).

Sure, when you print off a smartfridge board it will always be the base type fridge, but that's fine since you might actually want to build that one.